e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -0,0 +1,477 @@
+param(
+    [Parameter(Mandatory)]
+    [string] $Region,
+
+    [Parameter(Mandatory)]
+    [string] $ResourceGroup,
+    
+    [Parameter(Mandatory)]
+    [string] $SubscriptionId,
+
+    [Parameter(Mandatory)]
+    [string] $GroupCertificatePassword,
+
+    # Set this to true on the first execution to get everything installed in poweshell. Does not need to be run everytime.
+    [Parameter()]
+    [bool] $InstallDependencies = $true
+)
+
+$startTime = (Get-Date)
+
+########################################################################################################
+# Set error and warning preferences for the script to run
+########################################################################################################
+
+$ErrorActionPreference = "Stop"
+$WarningActionPreference = "Continue"
+
+###########################################################################
+# Connect-AzureSubscription - gets current Azure context or triggers a 
+# user log in to Azure. Selects the Azure subscription for creation of 
+# the virtual machine
+###########################################################################
+
+Function Connect-AzureSubscription()
+{
+    # Ensure the user is logged in
+    try
+    {
+        $azureContext = az account show
+    }
+    catch {
+    }
+
+    if (-not $azureContext)
+    {
+        Write-Host "`nPlease login to Azure..."
+        az login
+        $azureContext = az account show
+    }
+
+    # Ensure the desired subscription is selected
+    if ($azureContext.id -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+#################################################################################################
+# Set required parameters
+#################################################################################################
+
+$Region = $Region.Replace(' ', '')
+
+$StorageAccountName = "$($ResourceGroup.ToLower())sa"
+$HubName = $ResourceGroup
+$AppRegistrationName = $ResourceGroup
+$DeviceProvisioningServiceName = $ResourceGroup
+$FarRegion = "southeastasia"
+$FarHubName = $ResourceGroup + "Far"
+$UploadCertificateName = "group1-certificate"
```

Comment:
I noticed this is hard coded. Is that intentional?

Reply:
yes the tests expect this exact name